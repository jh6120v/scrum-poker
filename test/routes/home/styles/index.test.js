import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';
import {
    Container,
    Logo,
    Title,
    LogoLink
} from '../../../../src/routes/home/styles';

describe('test home styles', () => {
    it('match <Container /> snapshot', () => {
        const { container } = render(<Container />);
        const tree = container.firstChild;
        expect(tree).toMatchSnapshot();
    });

    it('match <Logo /> snapshot', () => {
        const { container } = render(<Logo />);
        const tree = container.firstChild;
        // const tree = renderer.create(<Logo />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('match <Title /> snapshot', () => {
        const { container } = render(<Title />);
        const tree = container.firstChild;
        // const tree = renderer.create(<Title />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('match <LogoLink /> snapshot', () => {
        const { container } = render(<LogoLink />);
        const tree = container.firstChild;
        // const tree = renderer.create(<LogoLink />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('test <Logo /> style when prop background images', () => {
        const { container } = render(<Logo img="fakeImg" />);
        const tree = container.firstChild;

        expect(tree).toHaveStyleRule('width', 'calc((100% - 100px) / 2)');
        expect(tree).toHaveStyleRule('height', '200px');
        expect(tree).toHaveStyleRule('background', 'url(fakeImg) no-repeat center');
        expect(tree).toHaveStyleRule('background-size', 'contain');
        expect(tree).toHaveStyleRule('animation', 'iVXCSc 3s linear infinite');
        expect(tree).toHaveStyleRule('width', '220px', {
            media: '(min-width: 768px)'
        });
    });

    it('test <LogoLink /> style when render it', () => {
        const { container } = render(<LogoLink />);
        const tree = container.firstChild;

        expect(tree).toHaveStyleRule('width', '80px');
        expect(tree).toHaveStyleRule('margin', '0 10px');
        expect(tree).toHaveStyleRule('position', 'relative');
        expect(tree).toHaveStyleRule('content', '\'\'', {
            modifier: ':before'
        });
        expect(tree).toHaveStyleRule('width', '1rem', {
            modifier: ':before'
        });
        expect(tree).toHaveStyleRule('height', '1rem', {
            modifier: ':before'
        });
        expect(tree).toHaveStyleRule('background-color', '#62DAFB', {
            modifier: ':before'
        });
        expect(tree).toHaveStyleRule('border-radius', '50%', {
            modifier: ':before'
        });
        expect(tree).toHaveStyleRule('position', 'absolute', {
            modifier: ':before'
        });
        expect(tree).toHaveStyleRule('left', '0', {
            modifier: ':before'
        });
        expect(tree).toHaveStyleRule('top', 'calc(50% - 0.5rem)', {
            modifier: ':before'
        });
        expect(tree).toHaveStyleRule('transition', 'all 1s ease-in-out', {
            modifier: ':before'
        });
        expect(tree).toHaveStyleRule('animation', 'KoXpa 1.7s infinite', {
            modifier: ':before'
        });

        expect(tree).toHaveStyleRule('width', '2rem', {
            media: '(min-width: 480px)',
            modifier: ':before'
        });
        expect(tree).toHaveStyleRule('height', '2rem', {
            media: '(min-width: 480px)',
            modifier: ':before'
        });
        expect(tree).toHaveStyleRule('top', 'calc(50% - 1rem)', {
            media: '(min-width: 480px)',
            modifier: ':before'
        });
    });
});
