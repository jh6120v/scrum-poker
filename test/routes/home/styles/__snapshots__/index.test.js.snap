// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test home styles match <Container /> snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  height: 100%;
  padding: 30px 15px;
  background: #e2e2e2;
  background: -moz-linear-gradient(45deg,#e2e2e2 0%,#dbdbdb 50%,#d1d1d1 51%,#fefefe 100%);
  background: -webkit-linear-gradient(45deg,#e2e2e2 0%,#dbdbdb 50%,#d1d1d1 51%,#fefefe 100%);
  background: linear-gradient(45deg,#e2e2e2 0%,#dbdbdb 50%,#d1d1d1 51%,#fefefe 100%);
  -webkit-filter: progid:DXImageTransform.Microsoft.gradient( startColorstr='#e2e2e2',endColorstr='#fefefe',GradientType=1 );
  filter: progid:DXImageTransform.Microsoft.gradient( startColorstr='#e2e2e2',endColorstr='#fefefe',GradientType=1 );
}

<div
  class="c0"
/>
`;

exports[`test home styles match <Logo /> snapshot 1`] = `
.c0 {
  width: calc((100% - 100px) / 2);
  height: 200px;
  background: url() no-repeat center;
  background-size: contain;
  -webkit-animation: iVXCSc 3s linear infinite;
  animation: iVXCSc 3s linear infinite;
}

@media (min-width:768px) {
  .c0 {
    width: 220px;
  }
}

<div
  class="c0"
/>
`;

exports[`test home styles match <LogoLink /> snapshot 1`] = `
.c0 {
  width: 80px;
  margin: 0 10px;
  position: relative;
}

.c0:before {
  content: '';
  width: 1rem;
  height: 1rem;
  background-color: #62DAFB;
  border-radius: 50%;
  position: absolute;
  left: 0;
  top: calc(50% - 0.5rem);
  -webkit-transition: all 1s ease-in-out;
  transition: all 1s ease-in-out;
  -webkit-animation: KoXpa 1.7s infinite;
  animation: KoXpa 1.7s infinite;
}

@media (min-width:480px) {
  .c0:before {
    width: 2rem;
    height: 2rem;
    top: calc(50% - 1rem);
  }
}

<div
  class="c0"
/>
`;

exports[`test home styles match <Title /> snapshot 1`] = `
.c0 {
  width: 100%;
  font-size: 40px;
  text-align: center;
  padding: 30px 0;
}

<h1
  class="c0"
/>
`;
